pkg/schedule/operator/operator_controller.go

```diff
@@ -126,7 +126,7 @@ func (oc *Controller) Dispatch(region *core.RegionInfo, source string, recordOpS
 				recordOpStepWithTTL(op.RegionID())
 			}
 			if oc.RemoveOperator(op, Succeed) {
-				operatorWaitCounter.WithLabelValues(op.Desc(), "promote-success").Inc()
+				operatorCounter.WithLabelValues(op.Desc(), "promote-success").Inc()
```

Comment:
Could we pre-initialize these label values inside the operator struct to reduce the cost here?

Reply:
Here we need to use `op.Desc()` which cannot be pre-initialized.

Reply:
Maybe we could initialize it when creating the operator like:

```go
func NewOperator(desc, brief string, regionID uint64, regionEpoch *metapb.RegionEpoch, kind OpKind, approximateSize int64, steps ...OpStep) *Operator {
	return &Operator{
		desc:                  desc,
		promoteSuccessCounter: operatorCounter.WithLabelValues(desc, "promote-success")
	}
}
```

Reply:
hmm, what is the difference?

Reply:
Oops, I misunderstood the initialization process here, it's the same if we initialize the metrics each time we create a new operator here.